framework/rendering/subpass.h

```diff
@@ -40,217 +50,442 @@ struct alignas(16) Light
+{
+	if constexpr (bindingType == vkb::BindingType::Cpp)
+	{
+		return depth_stencil_state;
+	}
+	else
+	{
+		return reinterpret_cast<vkb::DepthStencilState &>(depth_stencil_state);
+	}
+}
 
-	ShaderSource vertex_shader;
+template <vkb::BindingType bindingType>
+inline const bool &Subpass<bindingType>::get_disable_depth_stencil_attachment() const
+{
+	return disable_depth_stencil_attachment;
+}
 
-	ShaderSource fragment_shader;
+template <vkb::BindingType bindingType>
+inline const ShaderSource &Subpass<bindingType>::get_fragment_shader() const
+{
+	return fragment_shader;
+}
 
-	DepthStencilState depth_stencil_state{};
+template <vkb::BindingType bindingType>
+inline void Subpass<bindingType>::set_color_resolve_attachments(std::vector<uint32_t> color_resolve)
+{
+	color_resolve_attachments = color_resolve;
+}
 
-	/**
-	 * @brief When creating the renderpass, pDepthStencilAttachment will
-	 *        be set to nullptr, which disables depth testing
-	 */
-	bool disable_depth_stencil_attachment{false};
+template <vkb::BindingType bindingType>
+inline void Subpass<bindingType>::set_depth_stencil_resolve_attachment(uint32_t depth_stencil_resolve)
+{
+	depth_stencil_resolve_attachment = depth_stencil_resolve;
+}
 
-	/**
-	 * @brief When creating the renderpass, if not None, the resolve
-	 *        of the multisampled depth attachment will be enabled,
-	 *        with this mode, to depth_stencil_resolve_attachment
-	 */
-	VkResolveModeFlagBits depth_stencil_resolve_mode{VK_RESOLVE_MODE_NONE};
+template <vkb::BindingType bindingType>
+inline void Subpass<bindingType>::set_debug_name(const std::string &name)
+{
+	debug_name = name;
+}
 
-	/// Default to no input attachments
-	std::vector<uint32_t> input_attachments = {};
+template <vkb::BindingType bindingType>
+inline void Subpass<bindingType>::set_disable_depth_stencil_attachment(bool disable_depth_stencil)
+{
+	disable_depth_stencil_attachment = disable_depth_stencil;
+}
 
-	/// Default to swapchain output attachment
-	std::vector<uint32_t> output_attachments = {0};
+template <vkb::BindingType bindingType>
+inline void Subpass<bindingType>::set_depth_stencil_resolve_mode(ResolveModeFlagBitsType mode)
+{
+	if constexpr (bindingType == vkb::BindingType::Cpp)
+	{
+		depth_stencil_resolve_mode = mode;
+	}
+	else
+	{
+		depth_stencil_resolve_mode = static_cast<vk::ResolveModeFlagBits>(mode);
+	}
+}
 
-	/// Default to no color resolve attachments
-	std::vector<uint32_t> color_resolve_attachments = {};
+template <vkb::BindingType bindingType>
+inline void Subpass<bindingType>::set_input_attachments(std::vector<uint32_t> input)
+{
+	input_attachments = input;
+}
 
-	/// Default to no depth stencil resolve attachment
-	uint32_t depth_stencil_resolve_attachment{VK_ATTACHMENT_UNUSED};
-};
+template <vkb::BindingType bindingType>
+inline void Subpass<bindingType>::set_output_attachments(std::vector<uint32_t> output)
+{
+	output_attachments = output;
```

Comment:
copy vector on each call.  check if move is correct.

Reply:
As above.